### PR TITLE
Core/Entities: add extra use of UPDATETYPE_CREATE_OBJECT2

### PR DESCRIPTION
**Pull request by sirikfoll**
Mixup of commits TrinityCore@533517b and  cmangos/mangos-wotlk@89c2c8a

**Changes proposed:**

Use UPDATETYPE_CREATE_OBJECT2 for newly created objects of type:
-  Player, Pet, Corpse, DynamicObject
-  Summoned Units and Vehicles
-  Summoned GameObject wich has player as summoner

**Target branch(es):** 3.3.5/master
- [x] 3.3.5
- [ ] master

### CODE REVIEW

**Comment by tkrokli** [comment]
if you want AppVeyor to show correct result, you need https://github.com/TrinityCore/TrinityCore/commit/5241c1ad71bade65f431ce9b1c4e142c8386bf2b (build fix)

**Comment by Keader** [comment]
nice, fix alot of animations

**Comment by Shauren** [comment]
I'm sure @r00ty-tc has something to say about this

**Comment by r00ty-tc** [comment]
Yeah, I did see this. But I was working at the time. 

I'm unsure whether we should take the cmangos commit as-is, or just go with what to me, seems to make sense. That is, Create2 is used when a new object is created (any kind) and the packet signifies the creation and Create1 is used when an existing object is being sent to players it has now become relevant to. 

Not getting round to testing all object types to see if it made sense (and matched sniffs) is the main reason mine is sitting as a commit on my own repo. My suspicion is that either way is likely to have the desired effect.

If we go with a port from cmangos, it's probably fine. However I really dislike the naming. I'd suggest changing "itsnewobject" to something a little more like similar names "isNewObject" perhaps.

**Comment by sirikfoll** [comment]
It's completely up to you, I didn't do any research about this, just got the ideas from existing code and changed it to work for all summons, because I need it for the work i'm doing in Kologarn 😄

File: src/server/game/Maps/Map.cpp [hunk 1]
```diff
@@ -646,7 +646,9 @@ bool Map::AddToMap(T* obj)
 
     //something, such as vehicle, needs to be update immediately
     //also, trigger needs to cast spell, if not update, cannot see visual
+    obj->SetItsNewObject(true);
```
Comment (tkrokli) [inline src/server/game/Maps/Map.cpp]: (removed the `/home/travis/build/TrinityCore/TrinityCore/` part from the error message)
```
src/server/game/Maps/Map.cpp:649:10:
 fatal error: no member named 'SetItsNewObject' in 'Corpse'
    obj->SetItsNewObject(true);
    ~~~  ^
src/server/game/Maps/Map.cpp:3635:32:
 note: in instantiation of function template specialization 'Map::AddToMap<Corpse>' requested here
template TC_GAME_API bool Map::AddToMap(Corpse*);
                               ^
1 error generated.
```

**Comment by r00ty-tc** [comment]
Yeah, to be honest I'm happy to go with this way, until we do some more research of our own. My belief is that a lot of the conditions that are there based on type are because you will simply see type1 or type2 more for that kind of object. 

For example the check for GO for owner is player. Of course sniffs will show that because if a player spawns a gameobject from spell etc, they will see the type 2, and then anyone coming into range subsequently will see type 1.

But until we prove it for sure, this does solve the main issue, which is the correct go/creature anims.

**Comment by jackpoz** [comment]
@r00ty-tc so this can be merged ?

**Pull request merged**